src/rulesets/Base/mapreduce.jl

```diff
@@ -59,7 +59,7 @@ end
 
 function rrule(::typeof(sum), x)
     function sum_pullback(ȳ)
-        return (NO_FIELDS, cast(ȳ))
+        return (NO_FIELDS, @thunk(fill(ȳ, size(x))))
```

Comment:
should we consider using `Fill` from `FillArrays` instead of `fill`?

Reply:
That's definitely a thing we're going to do, but at a later date, just to avoid the added complexity for now (we don't want to have to deal with any `FillArray`-specific issues that aren't really `ChainRules` things for the time being)